### DISCUSSION TITLE
SegmentedControl component

### DISCUSSION THREAD

**Discussion by cloud-walker** [ariakit/ariakit#2712] [General]
Hi! I'm in a quest to implement a SegmentedControl component (similar to: https://primer.style/design/components/segmented-control or https://mantine.dev/core/segmented-control/).

My guess is that it is simply a radio group with custom UI (or maybe a radio inside a toolbar?), but then I've found discordant takes on the web on which accessibility pattern should follow.

Reading the ariakit docs I've then experienced a bit of fragmentation on the subject, we got: Radio, FormRadio and Toolbars, I've no idea on which is the best to use.

The ideal implementation to me is:

1. like a radio group
2. it works well inside forms also (as it is probably a radio group)
3. you can tweak the select behaviour to make it more lazy when needed (like that? #2674)

I'm in the right direction? There are currently workarounds to achieve that?

Thanks in advance!

**Comment by diegohaz** [comment]
It could be either `Radio` or `Tab`, based on the context. Given that you mention forms, it's undoubtedly a radio group. So, yes, I believe you're heading in the right direction. I know @ItsJonQ has worked on this in earlier versions of Reakit/Ariakit.

**Reply by ItsJonQ** [reply]
@diegohaz Yup!

When I did it with Reakit, it was using `RadioGroup` and `Radio`.
Those components handled the selection + radio state.

You would then write custom UI to render the selection animation (etc...) like what [Mantine](https://mantine.dev/core/segmented-control/) did with theirs.

Here's my original code for setting this up with Reakit. Hope this helps!

https://github.com/ItsJonQ/g2/blob/main/packages/components/src/SegmentedControl/SegmentedControl.js

**Reply by cloud-walker** [reply]
thanks! @diegohaz eventually you can think of a potential workaround for the selectOnMove thing?

**Reply by diegohaz** [reply]
@cloud-walker Could you please create a feature request with your use case and motivation as mentioned in https://github.com/ariakit/ariakit/discussions/2674#discussioncomment-6565195? It's easier to track this way.

**Reply by cloud-walker** [reply]
https://github.com/ariakit/ariakit/issues/2716
